extensions/script/python/PythonCreator.h

```diff
@@ -121,41 +122,34 @@ class PythonCreator : public minifi::core::CoreComponent {
     }
 
     for (const auto &path : pathOrFiles) {
-      utils::file::FileUtils::addFilesMatchingExtension(logger_, path, ".py", classpaths_);
+      utils::file::addFilesMatchingExtension(logger_, path, ".py", classpaths_);
     }
   }
 
-  std::string getPackage(const std::string& basePath, const std::string& pythonscript) {
-    const auto script_directory = getPath(pythonscript);
-    const auto loc = script_directory.find_first_of(basePath);
-    if (loc != 0 || script_directory.size() <= basePath.size()) {
+  std::string getPackage(const std::string &basePath, const std::string &pythonscript) {
+    if (!minifi::utils::StringUtils::startsWith(pythonscript, basePath)) {
       return "";
     }
-    const auto python_dir = script_directory.substr(basePath.length() + 1);
-    auto python_package = python_dir.substr(0, python_dir.find_last_of("/\\"));
-    utils::StringUtils::replaceAll(python_package, "/", ".");
-    utils::StringUtils::replaceAll(python_package, "\\", ".");
+    const auto python_package_path = std::filesystem::relative(pythonscript, basePath).parent_path();
+    std::vector<std::string> path_elements{python_package_path.begin(), python_package_path.end()};
+    std::string python_package = minifi::utils::StringUtils::join(".", path_elements);
     if (python_package.length() > 1 && python_package.at(0) == '.') {
       python_package = python_package.substr(1);
     }
```

Comment:
I don't think this is needed any longer, as none of the path elements can be empty, and `StringUtils::join()` is not going to put a separator at the front.

Reply:
<img width="325" alt="image" src="https://user-images.githubusercontent.com/30227161/152344524-0b301617-53db-4b08-a947-d76fc88225a3.png">
on my local it does

Reply:
That is very strange; I get "h2o.h2o3.mojo".  There is probably a bug somewhere, but it was most likely already there before this change, so we don't have to fix it now.